src/engine/strat_engine/tests/util.rs

```diff
@@ -0,0 +1,70 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::path::PathBuf;
+
+use devicemapper::{DM, DmFlags, DevId};
+
+use nix::mount::umount;
+use mnt::get_submounts;
+
+/// Remove all device mapper devices which have a name which matches the naming convention for
+/// a stratis created device.
+fn dm_stratis_devices_remove() {
+    match DM::new() {
+        Ok(dm) => {
+            // Lets assume that nesting won't go more than 10 deep...
+            // DM::table_deps returns a devices parent, not child which is what we would ideally
+            // want to systematically remove the devices in the correct order.
+            for _ in 0..10 {
+                let mut remain = 0;
+
+                match dm.list_devices() {
+                    Ok(devices) => {
+                        for d in devices {
+                            if format!("{}", d.0.as_ref()).starts_with("stratis-1-") {
+                                if let Err(_) = dm.device_remove(&DevId::Name(&d.0),
+                                                                 DmFlags::empty()) {
+                                    remain += 1;
+                                }
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        eprintln!("Error on list_devices() {:?}", e);
+                        break;
+                    }
+                }
+
+                if remain == 0 {
+                    break;
+                }
+            }
+        }
+        Err(e) => eprintln!("Error on DM::new() {:?}", e),
+    }
+}
+
+/// When a unit test panics we can leave the system in an inconsistent state.  This function
+/// tries to clean up by un-mounting any mounted file systems which contain the string
+/// "stratis_testing" and then it removes any device mapper tables which are also stratis created.
+pub fn clean_up() -> () {
+    match get_submounts(&PathBuf::from("/")) {
```

Comment:
You could make this block into a separate function, like ```unmount_stratis_filesystems```.

You will want to check that all filesystems made by tests are actually so conveniently named. I don't think I've wriiten too many filesystem tests, but my name choices tend to be completely random.

If a stratis filesystem can't be unmounted isn't the next step, getting rid of the devicemapper devices, at least partially doomed?